### PR TITLE
 core: vm: sqlvm: storage: implement bitmap and increasePK

### PR DESCRIPTION
**Pull request by jm-cobinhood**

### CODE REVIEW

File: core/vm/sqlvm/common/storage.go [hunk 1]
```diff
@@ -534,93 +534,142 @@ func (s *Storage) IncSequence(
...
-// UpdateHash updates hash to stateDB.
-func (s *Storage) UpdateHash(m map[common.Hash]common.Hash, address common.Address) {
-	for key, val := range m {
-		s.SetState(address, key, val)
-	}
+func (bm *bitMap) increasePK() uint64 {
+	lastRowID, rowCount := bm.decodeHeader()
+	lastRowID++
+	rowCount++
+	bm.headerData = bm.encodeHeader(lastRowID, rowCount)
```
Comment (yenlinlai) [inline core/vm/sqlvm/common/storage.go]: bm.headerData already assigned? also it seems no need of returning header?

File: core/vm/sqlvm/common/storage.go [hunk 2]
```diff
@@ -534,93 +534,142 @@ func (s *Storage) IncSequence(
...
+	shift := lastRowID/256 + 1
+	slot := bm.storage.ShiftHashUint64(bm.headerSlot, shift)
+	data := bm.storage.GetState(bm.address, slot)
+	byteShift := (lastRowID & 255) / 8
+	data[byteShift] |= 1 << (lastRowID & 7)
```
Comment (yenlinlai) [inline core/vm/sqlvm/common/storage.go]: maybe use setBit like the one in setPK?

File: core/vm/sqlvm/common/storage.go (hunk 1 above)
Comment (jm-cobinhood) [reply core/vm/sqlvm/common/storage.go]: fixed.

File: core/vm/sqlvm/common/storage.go (hunk 2 above)
Comment (jm-cobinhood) [reply core/vm/sqlvm/common/storage.go]: fixed.

**Pull request merged**